### PR TITLE
Allow getblocktemlate for not connected regtest node

### PR DESCRIPTION
**Pull request by ondrejsika**

### CODE REVIEW

**Comment by laanwj** [comment]
This has come up before in a discussion recently. 

However, instead of a special-case for regtest I think we should add a command line option to ignore the `getblocktemplate` pre-checks, both this one and the IsInitialBlockDownload - this makes it more general, and avoids switching on the name of the network, which is considered a bad thing.

**Comment by paveljanik** [comment]
Shouldn't regtest be "always connected"?

**Comment by MarcoFalke** [comment]
> 

Shouldn't regtest be "always connected"?


Probably not, it depends on your use case. So I agree with @laanwj and we should keep the special cases of regtest to a minimum.

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -465,7 +465,8 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
     if(!g_connman)
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
 
-    if (g_connman->GetNodeCount(CConnman::CONNECTIONS_ALL) == 0)
+    // Allow getblocktemlate for not connected regtest node
+    if (Params().NetworkIDString() != "regtest" && g_connman->GetNodeCount(CConnman::CONNECTIONS_ALL) == 0)
```
Comment (luke-jr) [inline src/rpc/mining.cpp]: Do the regtest comparison second so it can't incur *any* penalty on normal operation.

**Comment by fanquake** [comment]
Closing this due to inactivity. @ondrejsika if this is something your still working on/would like merged could you address the feedback here.

**Pull request closed without merging**